tx.go

```diff
@@ -168,6 +188,43 @@ func (tx *tx) CreateIterators(stmt *influxql.SelectStatement) ([]influxql.Iterat
 	return itrs, nil
 }
 
+// DecodeValues is for use in a raw data query
+func (tx *tx) DecodeValues(fieldIDs []uint8, timestamp int64, data []byte) []interface{} {
+	vals := make([]interface{}, len(fieldIDs)+1)
+	vals[0] = timestamp
+	for i, id := range fieldIDs {
+		v, _ := tx.decoder.DecodeByID(id, data)
+		vals[i+1] = v
+	}
+	return vals
+}
+
+// FieldIDs will take an array of fields and return the id associated with each
+func (tx *tx) FieldIDs(fields []*influxql.Field) ([]uint8, error) {
+	names := tx.fieldNames(fields)
```

Comment:
`Name` could be the `as` name though. I want the actual field name that was referenced, which this gives me.